pkg/kubefed2/federate/federate.go

```diff
@@ -183,32 +185,68 @@ func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName c
 	}, nil
 }
 
-func lookupTypeDetails(hostConfig *rest.Config, typeName ctlutil.QualifiedName) (*fedv1a1.FederatedTypeConfig, error) {
-	apiResource, err := enable.LookupAPIResource(hostConfig, typeName.Name, "")
+func getTypeConfig(hostConfig *rest.Config, typeName, namespace string, outputYAML bool) (typeconfig.Interface, error) {
+	// Lookup kubernetes API availability
+	apiResource, err := enable.LookupAPIResource(hostConfig, typeName, "")
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "Failed to find target API resource %s", typeName)
 	}
-	typeName.Name = typeconfig.GroupQualifiedName(*apiResource)
+	glog.V(2).Infof("API Resource for %s found", typeName)
 
-	client, err := genericclient.New(hostConfig)
+	typeConfig, err := attemptToLookupFederatedType(hostConfig, *apiResource, namespace)
+	if err == nil {
+		return typeConfig, nil
+	}
+	if apierrors.IsNotFound(err) && !outputYAML {
+		return nil, errors.Errorf("%v. Try 'kubefed2 enable type %s' before federating the resource", err, typeName)
```

Comment:
Please prefer Wrapf for instances like this:

errors.Wrapf(err, "Try kubefed2...", typeName)

Reply:
I did intentionally replace `Wrapf` with `Errorf`  because an error as below (actual output):
`error: federatedtypeconfigs.core.federation.k8s.io "deployments.apps" not found. Try 'kubefed2 enable type deployments.apps' before federating the resource`
reads better compared to
`error: Try 'kubefed2 enable type deployments.apps before federating the resource because federate failed with: resourcefederatedtypeconfigs.core.federation.k8s.io "deployments.apps" not found`